### DISCUSSION TITLE
Fixing MassTransit->Quartz integration

### DISCUSSION THREAD

**Discussion by sergeyshaykhullin** [quartznet/quartznet#1004] [Ideas] [answered]
The problem is in `ZeroSizeThreadPool` and `MassTransit` way to always-start schedulers

For now i have two solutions:
- Configure `MassTransit` integration to not to start external schedulers (Somehow(action/strategy, flag, whatever), idea is not complete)
- Ignore scheduler `Start/StandBy/Shutdown` if `ThreadPool` is `ZeroSizeThreadPool` =(

I want to discuss and complete more consistent solution:
- Is not obvious that with `ZeroSizeThreadPool` Start should not be called
- Is not obvious that with `ZeroSizeThreadPool` when Start is called nothing happend
- Is not obvious that `MassTransit` is checking `Quartz` `ThreadPool` type and desiding to not to start
- Is not obvious that `MassTransit` is checking some `Quartz` flags to deside should it Start scheduler or not (Also is not a good idea to have a lot of flags for whatever it needs. It is complicating configuration)

I think that `Quartz` has more potential for clean solution than `MassTransit.QuartzIntegration`, because of available internals

Related:
https://github.com/quartznet/quartznet/issues/1002
https://github.com/quartznet/quartznet/pull/1003
https://github.com/MassTransit/MassTransit/issues/2092

**Comment by lahma** [comment, marked as answer]
Created to https://github.com/MassTransit/MassTransit/pull/2095 to show one option how to configure this. I don't think there's much to do on Quartz's side.
